UseCases.md

```diff
@@ -0,0 +1,100 @@
+This file describes exact behavior of methods for different edge cases and 
+explains general logic. This description covers the behavior of get_tld, 
+get_tld_unsafe, get_sld, get_sld_unsafe, split_domain, split_domain_unsafe
+
+1. general difference of get_*() and get_*_unsafe() methods:
+get_*_unsafe() does not perform if the input string is None and does not 
+transforms it to the lower case.
+
+2. The listed above methods works only with non-canonical FQDN strings - 
+trailing dot must be removed before call the method. This restriction allows 
+get rid of fuzzy logic in edge cases.
+
+3. DNS does not support empty labels - if some label detected to be empty, 
+None will be returned. 
+
+4. Every method processes provided FQDN in the reverse order, from the last 
+label towards the start of the string. It stops when the specific task is 
+completed. Therefore no validation occurs outside of this scope.  For example,
+```
+get_tld('......com') -> 'com'
+``` 
+as leading dots are not processed.
+split_domain method is based on get_sld method - it returns everything in 
+front of get_sld() as a prefix.
+Specifically to example above 
+```
+split_domain('......com') -> ('....',None,'com')
+```
+Edge cases and expected behavior
+The behavior of the library can be illustrated best on the small examples:
+(boolean arguments are omitted if does not affect behavior )
+
```

Comment:
Suggested additional: 
The library allows you to create a public suffix list from any file-like object, including a list. In the examples below, we construct the PSL with different lists to demonstrate the functionality of the library under different conditions.  In particular, the examples show an empty list, list of two elements, and the use of negation in a list. Combinations of the method parameters are given with notes about the result.  These results follow the logic of the Mozilla library as published on the psl.org website.